src/main/scala/org/galaxio/gatling/kafka/client/KafkaSender.scala

```diff
@@ -1,41 +1,52 @@
 package org.galaxio.gatling.kafka.client
 
-import org.apache.kafka.clients.producer.{KafkaProducer, Producer, RecordMetadata}
-import org.galaxio.gatling.kafka.request.KafkaProtocolMessage
+import io.gatling.core.CoreComponents
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import org.galaxio.gatling.kafka.KafkaLogging
 
-import scala.concurrent.{ExecutionContext, Future}
+import java.util.concurrent.Executors
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService, Future}
 import scala.jdk.CollectionConverters._
 import scala.util.{Failure, Success}
 
-trait KafkaSender {
-  val executionContext: ExecutionContext
-  def send(protocolMessage: KafkaProtocolMessage)(
+trait KafkaSender[K, V] {
```

Comment:
We don't need type parameters here because it is only avro4s requirement. I use Array[Byte] for all types of keys and values for unification of publishing internal processing